runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java

```diff
@@ -354,7 +354,37 @@ public void onClientTick(ClientTick event)
 		}
 
 		Collections.reverse(newEntries);
-
+		
+		newEntries.sort((a, b) ->
+		{
+			final MenuAction aMenuType = a.getEntry().getType();
+			if (aMenuType == MenuAction.GROUND_ITEM_FIRST_OPTION || aMenuType == MenuAction.GROUND_ITEM_SECOND_OPTION || aMenuType == MenuAction.GROUND_ITEM_THIRD_OPTION
+				|| aMenuType == MenuAction.GROUND_ITEM_FOURTH_OPTION || aMenuType == MenuAction.GROUND_ITEM_FIFTH_OPTION || aMenuType == MenuAction.EXAMINE_ITEM_GROUND
+				|| aMenuType == MenuAction.WALK)
+			{
+				final MenuAction bMenuType = b.getEntry().getType();
+				if (bMenuType == MenuAction.GROUND_ITEM_FIRST_OPTION || bMenuType == MenuAction.GROUND_ITEM_SECOND_OPTION || bMenuType == MenuAction.GROUND_ITEM_THIRD_OPTION
+					|| bMenuType == MenuAction.GROUND_ITEM_FOURTH_OPTION || bMenuType == MenuAction.GROUND_ITEM_FIFTH_OPTION || bMenuType == MenuAction.EXAMINE_ITEM_GROUND
+					|| bMenuType == MenuAction.WALK)
+				{
+					final MenuEntry aEntry = a.getEntry();
+					final int aId = aEntry.getIdentifier();
+					final int aQuantity = getCollapsedItemQuantity(aId, aEntry.getTarget());
+					
+					final MenuEntry bEntry = b.getEntry();
+					final int bId = bEntry.getIdentifier();
+					final int bQuantity = getCollapsedItemQuantity(bId, bEntry.getTarget());
+					
+					if (config.sortByGEPrice())
```

Comment:
Wouldn't it be better to have this check at the beginning of the sort, and return if the option isn't enabled? If the user doesn't have this option enabled, we're making some unnecessary calculations.